crates/solver/src/solver/single_order_solver.rs

```diff
@@ -1167,8 +1167,8 @@ mod tests {
             TradeExecution {
                 sell_token,
                 buy_token,
-                sell_amount: 52.into(),
-                buy_amount: 60.into(),
+                sell_amount: 52.into(), // 2 solver fee
+                buy_amount: 57.into(), // 3 protocol fee
                 fee_amount: 0.into(),
```

Comment:
This is a weird one. Sell partial limit order. Solver executed 50% of sell_amount=100. 
Solver fee increases sell_amount.
But for protocol fee we always reduce buy amount for sell orders. This is so that we don't have to change anything in solver behavior to include protocol fees.